nvi-rest/src/test/java/no/sikt/nva/nvi/rest/fetch/FetchReportHandlerTest.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+class FetchReportHandlerTest extends LocalDynamoTest {
+
+    private static final Context CONTEXT = mock(Context.class);
+    private final DynamoDbClient localDynamo = initializeTestDatabase();
+    private ByteArrayOutputStream output;
+    private FetchReportHandler handler;
+    private CandidateRepository candidateRepository;
+    private PeriodRepository periodRepository;
+
+    @BeforeEach
+    public void setUp() {
+        output = new ByteArrayOutputStream();
+        candidateRepository = new CandidateRepository(localDynamo);
+        periodRepository = periodRepositoryReturningOpenedPeriod(CURRENT_YEAR);
+        handler = new FetchReportHandler(candidateRepository, periodRepository);
+    }
+
+    @Test
+    void shouldReturnUnauthenticatedWhenNotLoggedInWithValidUser() throws IOException {
+        var identifier = randomUri();
+        var customerId = randomUri();
+        var year = Year.now().getValue();
+        var input = createInputWithoutAccessRights(customerId, identifier, year).build();
+        handler.handleRequest(input, output, CONTEXT);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_UNAUTHORIZED)));
+    }
+
+    @Test
+    void shouldReturnNotFoundIfInstitutionDoesntExist() {
+
+    }
+
+    @Test
+    void shouldReturnEmptyXlsxFileWhenNotDataExists()
+        throws IOException {
+        var input = createInput(randomUri(), randomUri(), Year.now().getValue())
+                        .build();
+        handler.handleRequest(input, output, CONTEXT);
+        var response = GatewayResponse.fromOutputStream(output, String.class);
+
+        assertThat(response.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+        List<List<String>> data = readFromExcel(response.getBody());
+
+        assertThat(response.getBody(), is(notNullValue()));
+        assertThat(data.get(0).size(), is(ReportRow.class.getRecordComponents().length));
+    }
+
+    @Test
+    void shouldContentNegotiate() throws IOException {
+        var input = createInput(randomUri(), randomUri(), Year.now().getValue())
+                        .withHeaders(Map.of(HttpHeaders.ACCEPT, MediaType.MICROSOFT_EXCEL.toString()))
+                        .build();
+        handler.handleRequest(input, output, CONTEXT);
+        var response = GatewayResponse.fromOutputStream(output, String.class);
+        assertThat(response.getHeaders().get(HttpHeaders.CONTENT_TYPE), is(MediaType.MICROSOFT_EXCEL.toString()));
+    }
+
+    @Test
+    void shouldContentNegotiateMore() throws IOException {
```

Comment:
Rename to `shouldReturnContentTypeExcelWhenAnyContentTypeIsAccepted()` or something?